metadata-io/src/main/java/com/linkedin/metadata/search/elasticsearch/query/ESSearchDAO.java

```diff
@@ -335,9 +335,10 @@ public ScrollResult scroll(@Nonnull List<String> entities, @Nonnull String input
 
   private static Criterion transformEntityTypeCriterion(Criterion criterion, IndexConvention indexConvention) {
     return criterion.setField("_index").setValues(
-        new StringArray(criterion.getValues().stream().map(
-            indexConvention::getEntityIndexName).collect(
-            Collectors.toList()))).setValue(indexConvention.getEntityIndexName(criterion.getValue()));
+        new StringArray(criterion.getValues().stream().map(value -> String.join("", value.split("_")))
+            .map(indexConvention::getEntityIndexName)
```

Comment:
nice! would it make sense to put this logic in `getEntityIndexName`? is there anywhere else that calls this method that needs this logic around `_`?

Reply:
I didn't want to mess with that since it is widely used by other functions, and I didn't want to break those usages.

Reply:
and the other places this is used don't require this logic though?